program/src/instructions/withdraw_sol.rs

```diff
@@ -36,10 +37,19 @@ impl<'info> WithdrawSol<'info> {
 }
 
 impl<'info> WithdrawSol<'info> {
-    fn validate_sol_transfer(&self) -> Result<()> {
+    fn validate(&self) -> Result<()> {
         if self.pool.shared_escrow != Pubkey::default() {
             throw_err!(ErrorCode::PoolOnSharedEscrow);
         }
+        match self.pool.config.pool_type {
+            PoolType::NFT | PoolType::Trade => (),
```

Comment:
Ooof yea here is where redundancy actually introduced a bug. 

Should be Token || Trade.